### ISSUE TITLE
Black Text on Darker Background

### ISSUE THREAD

**Issue by ghost** [imatimba/darker-chromium#1]
First of all, thanks so much for this.

There is one thing bugging me though, the background is so dark that it makes black text hard to read. you can go to about:plugins to see an example of this.

**Comment by ghost** [comment]
Someone left this suggestion on the bug tracker

>Comment 241 by ryalia...@gmail.com, Today (6 hours ago)
>I'm a beginner computer programmer. 
>
>Would just implementing body { background-color: black; background-image: url(whitebackgroundpicture.jpg) } work? 
>
>In my page, I've tried using this piece of code and I get a black flash before loading the image background.
>
>Would this help?

Something to try I suppose? Although whitebackgroundpicture.jpg should probably be more of a light gray...

**Comment by imatimba** [comment]
Hi, a lot of internal pages aren't themed yet, including chrome://plugins.
I'm working on it, you can see the current list of themed pages on the release notes.
Changing WebKit default background color already changes the background of most internal pages, but as you can see on plugins I need to edit the text color and other sections to make it readable.
I'll release an update this week probably.

**Comment by ghost** [comment]
You can't account for extensions, though.

**Comment by imatimba** [comment]
Exactly, as I explain on the readme changing WebKit default background color has that drawback.
Some extensions might become unreadable.
I'm not an experienced dev at the moment so it would be difficult for me to find an alternative solution to the white flash on new windows.
Hopefully Google will fix it soon enabling me to revert the change and implement the background color on each css/html.

**Comment by ghost** [comment]
Did you see the comment I quoted above? Could you try that?

**Comment by imatimba** [comment]
As far as I know the white flash on new windows occurs before loading the style.
But I'll test it to be sure.

**Comment by ghost** [comment]
Sorry, you're right. it does occur before loading the style, I was confused as to what the suggestion was.

I see now the commit that fixes the white flash is here: https://github.com/hbtlabs/chromium-white-flash-fix/commit/b1c4d8efd43fa46c61b9dc695f387f41a16947fb

**Comment by ghost** [comment]
Okay, just brain storming here, but is there any way to apply styles to WebKit windows (meaning, after the widget 'loads')?

That way you could leave the default WebKit background that dark color to get rid of the unnecessary white flash, but extension interfaces would still have a light background.

**Comment by imatimba** [comment]
I'm not sure, but I'll search the code to see if it's possible. Could be a good alternative.

**Comment by hbt** [comment]
One way to fix it is to load a custom CSS in the document. It would apply to all pages. 

View third_party/WebKit/Source/core/dom/Document.cpp:569
```
// working version with --no-sandbox

#include "base/bind.h"
#include "base/files/file_util.h"
#include <iostream>
#include <fstream>
#include <cstdlib>
#include <fstream>
#include <iostream>

  std::string css;
  base::ReadFileToString(base::FilePath("/tmp/r"), &css);
  std::cout << css;
  base::WriteFile(base::FilePath("/tmp/r3"), "ggas", strlen("ggas"));
  StyleSheetContents *parsedSheet = StyleSheetContents::create(CSSParserContext(*this, 0));
  String css2 = String::String(css.c_str());
  parsedSheet->parseString(css2);
  styleEngine().injectAuthorSheet(parsedSheet);

```

However, other than hardcoding the css string or using `--sandbox` to load a custom stylesheet, I could not find an easy way to do it.

**Comment by ghost** [comment]
What is wrong with hardcoding the css string?

**Comment by hbt** [comment]
@Lady-Luck 
You have to get the CSS right so it works for all sites or you will need a stylesheet toggler extension (to fix it for domains where it doesn't work)

**Comment by ghost** [comment]
How's the next update coming along?

**Comment by imatimba** [comment]
I've completed a few extra pages, but I'm gonna wait for the next stable release that should be out in the next days to avoid compiling and releasing two times in a week.
In the meantime I'm setting up a few chroots to build Linux packages because I can't get to work the Linux installers gn target.

Edit: nvm, the stable update is out, I'll see if can release the update tomorrow.
